clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -161,7 +162,9 @@
         new NotControllerException("This is not the correct controller for this cluster.")),
     INVALID_REQUEST(42,
         new InvalidRequestException("This most likely occurs because of a request being malformed by the client library or" +
-            " the message was sent to an incompatible broker. See the broker logs for more details."));
+            " the message was sent to an incompatible broker. See the broker logs for more details.")),
+    UNSUPPORTED_FOR_MESSAGE_FORMAT(43,
```

Comment:
We should probably update the possible error codes documented in `ListOffsetResponse`.